python/cudf/cudf/core/window/rolling.py

```diff
@@ -560,31 +586,39 @@ def __init__(self, groupby, window, min_periods=None, center=False):
             sort_order
         )
 
-        gb_size = groupby.size().sort_index()
-        self._group_starts = (
-            gb_size.cumsum().shift(1).fillna(0).repeat(gb_size)
-        )
+        if not is_integer(window):
+            gb_size = groupby.size().sort_index()
+            self._group_starts = (
```

Comment:
I found it a bit hard to follow why `_group_starts` is getting set this way, and that led me down a bit of a rabbit hole. Do you think it would be cleaner to move input validation from `__init__` into separate helper functions so that `RollingGroupby` and `Rolling` could have completely separate `__init__` functions that call helpers, and then inline the code for converting windows to sizes? The current way that `_normalize` and `_window_to_window_sizes` is set up seems to improve code reuse but at the significant expense of readability.

Out of scope for this PR, but a suggestion for future improvement.

Reply:
Yeah agreed there's an opportunity here to make things clearer. Additionally, we may not even need this if we can convert this to purely use pylibcudf as described in https://github.com/rapidsai/cudf/pull/19162#discussion_r2167244584